examples/cross_section.py

```diff
@@ -0,0 +1,135 @@
+Cross Section Analysis
+======================
+
+The MetPy function `metpy.interpolate.cross_section` can obtain a cross-sectional slice through
+gridded data.
+"""
+
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+
+import metpy.calc as mpcalc
+from metpy.cbook import get_test_data
+from metpy.interpolate import cross_section
+
+##############################
+# **Getting the data**
+#
+# This example uses NARR reanalysis data for 18 UTC 04 April 1987 from NCEI
+# (https://www.ncdc.noaa.gov/data-access/model-data).
+#
+# We use MetPy's CF parsing to get the data ready for use, and squeeze down the size-one time
+# dimension.
+
+data = xr.open_dataset(get_test_data('narr_example.nc', False))
+data = data.metpy.parse_cf().squeeze()
+print(data)
+
+##############################
+# Define start and end points:
+
+start = (37.0, -105.0)
+end = (35.5, -65.0)
+
+##############################
+# Get the cross section, and convert lat/lon to supplementary coordinates:
+
+cross = cross_section(data, start, end)
+cross.set_coords(('lat', 'lon'), True)
+print(cross)
+
+##############################
+# For this example, we will be plotting potential temperature, relative humidity, and
+# tangential/normal winds. And so, we need to calculate those, and add them to the dataset:
+
+temperature, pressure, specific_humidity = xr.broadcast(cross['Temperature'],
+                                                        cross['isobaric'],
+                                                        cross['Specific_humidity'])
+
+theta = mpcalc.potential_temperature(pressure, temperature)
+rh = mpcalc.relative_humidity_from_specific_humidity(specific_humidity, temperature, pressure)
+
+# These calculations return unit arrays, so put those back into DataArrays in our Dataset
+cross['Potential_temperature'] = xr.DataArray(theta,
+                                              coords=temperature.coords,
+                                              dims=temperature.dims,
+                                              attrs={'units': theta.units})
+cross['Relative_humidity'] = xr.DataArray(rh,
+                                          coords=specific_humidity.coords,
+                                          dims=specific_humidity.dims,
+                                          attrs={'units': rh.units})
+
+cross['u_wind'].metpy.convert_units('knots')
+cross['v_wind'].metpy.convert_units('knots')
+cross['t_wind'], cross['n_wind'] = mpcalc.cross_section_components(cross['u_wind'],
+                                                                   cross['v_wind'])
+
+print(cross)
+
+##############################
+# Now, we can make the plot.
+
+# Define the figure object and primary axes
+fig = plt.figure(1, figsize=(16., 9.))
+ax = plt.axes()
+
+# Plot RH using xarray's contourf wrapper
+cross['Relative_humidity'].plot.contourf(x='lon', y='isobaric', yincrease=False, ax=ax,
```

Comment:
Is it any (much?) worse to use `ax.contour`? I really hate this way of using xarray. IMO, my data should not be plotting anything.

Reply:
Sorry, I was not aware of that! (I'll be sure to avoid that in the other examples/demos I'm working on). It would be a little bit more verbose (as I would have to use `cross['lon']` instead of `'lon'`, and manually specify labels), but not that bad of change.

Reply:
No worries, this may just be a case of me yelling at the damn kids to get off my lawn, but I'm willing to hold out. 😁